### PR TITLE
Fixes #17028,#17029 - use candlepin 2.0 api

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
[test]

**Comment by jlsherrill** [comment]
[test]

**Comment by daviddavis** [comment]
Just curious about why RefreshSubscriptions was removed.

Other than that, tested it out and everything worked. So ACK.

**Comment by jlsherrill** [comment]
@daviddavis RefreshSubscription was removed because it no longer did anything:

http://www.candlepinproject.org/docs/candlepin/cp2_migration.html

> Refresh pools API is no longer relevant in Satellite will no-op. Uses of it should be removed.

It actually caused a problem because it was no longer returning a valid json body.

**Comment by ehelms** [comment]
@jlsherrill is this mergeable yet? Looks like @daviddavis approved it and it could use a rebase.

File: app/lib/actions/candlepin/product/update.rb [hunk 1]
```diff
@@ -18,6 +20,7 @@ def plan(product)
                                          :label => pc.content.label,
```
Comment (bbuckingham) [inline app/lib/actions/candlepin/product/update.rb]: Should this call to ::Actions::Candlepin::Product::ContentCreate include the owner?

File: app/lib/actions/katello/product/content_create.rb [hunk 2]
```diff
@@ -8,12 +8,14 @@ def plan(repository)
...
+                                           owner:       repository.product.organization.label,
                                            name:        repository.name,
                                            type:        repository.content_type,
                                            label:       repository.custom_content_label,
                                            content_url: content_url(repository))
               content_id = content_create.output[:response][:id]
               plan_action(Candlepin::Product::ContentAdd,
+                                    owner: repository.product.organization.label,
                                     product_id: repository.product.cp_id,
                                     content_id: content_id)
```
Comment (bbuckingham) [inline app/lib/actions/katello/product/content_create.rb]: Should the call to Candlepin::Product::ContentUpdate in this action include the owner?

**Comment by bbuckingham** [comment]
Attempting to remove a subscription from a content host using the UI generated the following error:

```
 An error occurred removing the subscriptions.Task 3cc327b0-4abb-460e-b2d6-bf069d8efb32: NoMethodError: undefined method `with_indifferent_access' for #<Array:0x007f56edd8d2b8>; NoMethodError: undefined method `with_indifferent_access' for #<Array:0x007f56f43fa460>
```

Full stack: https://gist.github.com/bbuckingham/9667642fe0a873535481efdcf3cf69dd

File: app/lib/actions/katello/product/content_create.rb (hunk 2 above)
Comment (jlsherrill) [inline app/lib/actions/katello/product/content_create.rb]: good catch!

File: app/lib/actions/candlepin/product/update.rb (hunk 1 above)
Comment (jlsherrill) [inline app/lib/actions/candlepin/product/update.rb]: good catch too

**Comment by bbuckingham** [comment]
@jlsherrill, reviewed, ran several tests with the latest candlepin 2.0 build (22) and did not encounter any issues.   ACK from me.

**Comment by jlsherrill** [comment]
[test]

**Comment by jlsherrill** [comment]
Plan to merge as soon as nightly is working, hopefully in the next few days

**Pull request merged**